sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -404,6 +404,28 @@ private[hive] class HiveClientImpl(
     Option(hivePartition).map(fromHivePartition)
   }
 
+  /**
+   * Returns the partition names from hive metastore for a given table in a database.
+   */
+  override def getPartitionNames(
+      db: String,
+      table: String,
+      range: Short): Seq[String] = withHiveState {
```

Comment:
Sorry, i should be passing the range to hive's API. I had it hard coded to -1 initially and later changed , so that caller of this API can pass in a range to restrict the number of partitions that are returned to the client. I will make the change to pass the supplied range to hive.